### PR TITLE
fixes for Django 4.0

### PR DESCRIPTION
**Pull request by PetrDlouhy**

### CODE REVIEW

**Comment by RevolutionTech** [comment]
Disclosure: I don't work at InfoScout anymore so this is just a drive-by review.

You could consider updating `.circleci/config.yml` and `tox.ini` so that Django 2.0-4.0 are all tested in CI. That might encourage the maintainers to merge this PR (though I can't speak for them).

**Comment by PetrDlouhy** [comment]
I have added the newer versions of Django/Python to the CircleCI configuration. All tests are passing now: https://app.circleci.com/pipelines/github/PetrDlouhy/django-cache-utils?branch=django40&filter=all

Although there is probably some problem in configuration, the tests results are not shown here.

**Comment by PetrDlouhy** [comment]
I have also added fix for Django 4.1 deprecation warning and changed tests, that they run with deprecation check.

**Comment by MilanZiaran** [comment]
Hello, what is the current progress with this PR? Support for Django 4.0 would be much appreciated.

File: setup.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def run(self):
         "Caching decorator and django cache backend with advanced invalidation ability and dog-pile effect prevention"
     ),
     long_description=open('README.md').read(),
-    install_requires=['Django >= 1.8', 'python_memcached'],
+    install_requires=['Django >= 1.8', 'python_memcached', 'pymemcache'],
```
Comment (RevolutionTech) [inline setup.py:52]: Should `pymemcache` replace `python_memcached`, or is there a reason `python_memcached` is still needed?

File: tox.ini [hunk 2]
```diff
@@ -1,15 +1,22 @@
 [tox]
-envlist = dj{18,19,110,111}
+envlist = dj{18,19,110,111;20;21;22;30;31;32;40}
 
 [testenv]
 passenv = CI CIRCLECI CIRCLE_*
 commands =
     pip install python-memcached coverage codecov
```
Comment (RevolutionTech) [inline tox.ini:7]: Does this need to be updated to use `pymemcache`?

**Comment by PetrDlouhy** [comment]
@RevolutionTech I tried to remove the `python_memcached` references, but the tests stopped passing in Django<3.2 with `ModuleNotFoundError: No module named 'memcache' `. So I expect that we should either drop older versions or keep the dependency.

**Review by cyolveren — APPROVED** [review]
👍  looks good

**Comment by RevolutionTech** [comment]
Thanks @cyolveren! :100:

**Pull request merged**